internal/cli/app_docs.go

```diff
@@ -270,6 +272,56 @@ func (c *AppDocsCommand) emitSection(w io.Writer, name, use, h string, fields []
 	}
 }
 
+// jsonFormat attempts to output all the data included in a a plugin's Documentation() function in the JSON file format
+func (c *AppDocsCommand) jsonFormat(name, ct string, doc *docs.Documentation) {
+	// we use this constant to compare to ct for some special behavior
+	const csType = "configsourcer"
+
+	w, err := os.Create(fmt.Sprintf("./docs/%s-%s.json", ct, name))
```

Comment:
My intention here is to always write the JSON-formatted documentation to waypoint/docs/. I believe that that path will reliably do that, I mimicked how its done in mdxFormat() on line 328.

Reply:
A suggestion here I had was to do something like `docs/gen/*.json` similar to how we store auto-generated grpc/protobufs in `pkg/server/gen`. Since these are machine generated, it would be clearer to those browsing the source that they shouldn't be modifying these json files directly.